File: SlooshIOS/neomovies-android-main/app/src/main/java/com/neo/neomovies/ui/util/FilterMovies.kt
package com.neo.neomovies.ui.util

import com.neo.neomovies.data.network.dto.MediaDto

fun filterValidMovies(items: List<MediaDto>): List<MediaDto> {
    return items.filter { item ->
        val hasPoster = !item.posterUrl.isNullOrBlank() &&
            !item.posterUrl.contains("no-poster", ignoreCase = true)

        val hasTitle = !item.title.isNullOrBlank() || !item.name.isNullOrBlank()

        val hasRating = item.rating != null && item.rating > 0.0

        hasPoster && hasTitle && hasRating
    }
}

File: SlooshIOS/neomovies-web-main/src/utils/filterMovies.ts
import type { Movie } from '../types'

export const filterValidMovies = (movies: Movie[]): Movie[] => {
  return movies.filter((movie) => {
    // Check if has poster
    const hasPoster = !!(
      movie.poster_path ||
      movie.posterUrlPreview ||
      movie.posterUrl
    )

    // Check if has title/name
    const hasTitle = !!(
      movie.title ||
      movie.name ||
      movie.nameRu ||
      movie.nameOriginal
    )

    // Year/rating in v2 API can be absent or equal to 0, so we do not
    // hard-filter by them; otherwise full pages become empty.
    return hasPoster && hasTitle
  })
}
